target/linux/ath79/image/generic.mk

```diff
@@ -999,6 +999,21 @@ define Device/librerouter_librerouter-v1
 endef
 TARGET_DEVICES += librerouter_librerouter-v1
 
+define Device/meraki_mr16
+  SOC := ar7161
+  DEVICE_VENDOR := Meraki
+  DEVICE_MODEL := MR16
+  ROOTFS_SIZE := 13440k
+  IMAGE_SIZE := 15680k
+  DEVICE_PACKAGES := kmod-spi-gpio kmod-owl-loader
+  IMAGE/kernel.bin := append-kernel
+  IMAGE/rootfs.bin := append-rootfs | pad-rootfs
+  IMAGE/sysupgrade.bin := append-rootfs | pad-rootfs | pad-to $$$$(ROOTFS_SIZE) | append-kernel | append-metadata | check-size
+  IMAGES := kernel.bin rootfs.bin sysupgrade.bin
```

Comment:
Defining these images is critical. We do kernel and rootfs separate for the u-boot flashing instructions, and we make our own sysupgrade to help us handle how the rootfs and kernel are configured in u-boot to boot.

Reply:
Good to know. I'll merge this now and I'd like to verify that sysupgrade (forced, due to different architecture) will work from 18.06 to OpenWrt master.